assets/src/edit-story/app/history/useHistoryReducer.js

```diff
@@ -42,9 +42,35 @@ const reducer = (size) => (state, { type, payload }) => {
         );
 
         if (isReplay) {
+          const currentEntry = state.entries[state.offset];
+          const offset = state.entries.indexOf(state.replayState);
+          // If a page has changed and it was not an added page, and if the page is about to change with replay,
+          // Ensure the user stays on the page where the latest change happened.
+          if (
+            currentEntry.pages !== state.replayState.pages &&
+            currentEntry.pages.length === state.replayState.pages.length &&
+            currentEntry.current !== state.replayState.current
+          ) {
+            const changedPage = currentEntry.pages.filter((page, index) => {
+              return page !== state.replayState.pages[index];
+            });
+            // If a changed page was found.
+            if (changedPage.length === 1) {
+              // Ensure we stay on the changed page by overriding the previously saved current page.
+              const current = changedPage[0].id;
```

Comment:
```suggestion
            if (changedPage) {
              // Ensure we stay on the changed page by overriding the previously saved current page.
              const current = changedPage.id;
```

Reply:
It should only do the thing if there was only one page changed, thus the check for the length.

Reply:
I think I got a little confused by the variable being named in singular while being an array. But that's a i tiny nit, so I'm good.